### PR TITLE
fix: detection logic, 3rd plugin

### PR DESCRIPTION
**Pull request by Rushaway**

### CODE REVIEW

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 1]
```diff
@@ -76,6 +86,31 @@ public void OnPluginStart()
 			}
 		}
 	}
+
+	g_bLate = false;
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: not needed

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 2]
```diff
@@ -350,30 +399,35 @@ void DoStats(CPlayer Player, CStreak CurStreak, CJump hJump)
 	CurStreak.SetJumps(aStreakJumps);
 
 	int iStreakJumps = CurStreak.iJumps;
-	if (iStreakJumps >= 6)
+	if (iStreakJumps >= 10)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: can we put these values in cvars

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 3]
```diff
@@ -350,30 +399,35 @@ void DoStats(CPlayer Player, CStreak CurStreak, CJump hJump)
...
 	{
 		float HackRatio = CurStreak.iHackJumps / float(iStreakJumps);
-		if (HackRatio >= 0.85 && !Player.bFlagged)
+		if (HackRatio >= 0.95 && !Player.bFlagged)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: can we put these values in cvars

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 4]
```diff
@@ -350,30 +399,35 @@ void DoStats(CPlayer Player, CStreak CurStreak, CJump hJump)
...
 				KickClient(client, "Turn off your hack!");
 				LogAction(-1, client, "[AntiBhopCheat] \"%L\" was kicked for using bhop hack streak.", client);
 			}
-			ResetValues(client);
 			return;
 		}
 
 		float HyperRatio = CurStreak.iHyperJumps / float(iStreakJumps);
-		if (HyperRatio >= 0.85 && !Player.bFlagged)
+		if (HyperRatio >= 0.95 && !Player.bFlagged)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: can we put these values in cvars

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 5]
```diff
@@ -383,27 +437,32 @@ void DoStats(CPlayer Player, CStreak CurStreak, CJump hJump)
 	if (iGlobalJumps >= 25)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: can we put these values in cvars

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 6]
```diff
@@ -383,27 +437,32 @@ void DoStats(CPlayer Player, CStreak CurStreak, CJump hJump)
...
 	{
 		float HackRatio = Player.iHackJumps / float(iGlobalJumps);
-		if (HackRatio >= 0.65 && !Player.bFlagged)
+		if (HackRatio >= 0.75 && !Player.bFlagged)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: can we put these values in cvars

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 7]
```diff
@@ -383,27 +437,32 @@ void DoStats(CPlayer Player, CStreak CurStreak, CJump hJump)
...
 				KickClient(client, "Turn off your hack!");
 				LogAction(-1, client, "[AntiBhopCheat] \"%L\" was kicked for using global bhop hack.", client);
 			}
-			ResetValues(client);
 			return;
 		}
 
 		float HyperRatio = Player.iHyperJumps / float(iGlobalJumps);
-		if (HyperRatio >= 0.50 && !Player.bFlagged)
+		if (HyperRatio >= 0.75 && !Player.bFlagged)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: can we put these values in cvars

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 8]
```diff
@@ -431,7 +490,13 @@ void NotifyAdmins(int client, const char[] sReason)
 		}
 	}
 
-	Forward_OnDetected(client, sReason, g_sStats);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: why did you remove the function

Comment (Rushaway) [reply addons/sourcemod/scripting/AntiBhopCheat.sp]: Making a function of something who is used only time is not usefull.

**Comment by Rushaway** [comment]
Cvar is the best way to go instead of hardcoding. Value have to be protected (anti cheats plugin..)

+Adding cvar for Limit Bhop & Kick for Current Streak & Global Streak

File: addons/sourcemod/scripting/AntiBhopCheat.sp (hunk 8 above)
Comment (maximee-leroyy) [reply addons/sourcemod/scripting/AntiBhopCheat.sp]: its a C++ convention that we should follow for better code readability:
- https://en.wikibooks.org/wiki/C%2B%2B_Programming/Programming_Languages/C%2B%2B/Code/Style_Conventions#25_lines_80_columns

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 9]
```diff
@@ -62,32 +77,76 @@ public void OnPluginStart()
...
+	HookConVarChange(g_cvGlobalHyper, OnConVarChanged);
+	HookConVarChange(g_cvGlobalHack, OnConVarChanged);
+	HookConVarChange(g_cvGlobalHackKick, OnConVarChanged);
+#if defined _SelectiveBhop_Included
+	HookConVarChange(g_cvCurrentLimitBhop, OnConVarChanged);
+	HookConVarChange(g_cvGlobalLimitBhop, OnConVarChanged);
+#endif
+
+	// Support for plugin reload
+	g_iSvGravity = GetConVarInt(g_cvSvGravity);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: Put it in OnConfigsExecuted

File: addons/sourcemod/scripting/AntiBhopCheat.sp [hunk 10]
```diff
@@ -1,69 +1,139 @@
...
+	HookConVarChange(g_cvGlobalHyper, OnConVarChanged);
+	HookConVarChange(g_cvGlobalHack, OnConVarChanged);
+	HookConVarChange(g_cvGlobalHackKick, OnConVarChanged);
+#if defined _SelectiveBhop_Included
+	HookConVarChange(g_cvCurrentLimitBhop, OnConVarChanged);
+	HookConVarChange(g_cvGlobalLimitBhop, OnConVarChanged);
+#endif
+
+	// Support for plugin reload
+	g_iMaxFlags = GetConVarInt(g_cvMaxDetections);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/AntiBhopCheat.sp]: i meant to move all GetConVar lines inside OnConfigsExecuted^^
